### PR TITLE
Added front end pagination to display recipes

### PR DESCRIPTION
**Pull request by Lord-Bubbles**
#14 #16 #17

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,8 +29,5 @@ yarn-error.log*
 
 .vscode
 
-/src/components/Pagination.js
-/src/components/Recipe.js
-/src/components/RecipeView.js
 
-MongoDB Certificate.pem
+
```
Comment (chilimalone) [inline .gitignore:33]: only need 1 newline at end of file

File: src/components/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    const leftSiblingIndex = Math.max(currentPage - siblings, 1);
+    const rightSiblingIndex = Math.min(currentPage + siblings, totalPages);
+
+    const showLeftDots = leftSiblingIndex > 2;
+    const showRightDots = rightSiblingIndex < totalPages - 2;
+
+    const firstPageIndex = 1;
+    const lastPageIndex = totalPages;
+
+    // Currentpage + firstpage + lastpage + 2 * numSiblings
```
Comment (chilimalone) [inline src/components/Pagination.js:42]: what does this even mean

File: src/components/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  handleRight = () => {
+    this.props.onPageChange(this.props.currentPage + 1);
+  }
+
+  fetchPageNumbers = () => {
+    const totalPages = Math.ceil(this.props.total / this.props.pageSize);
+    const {currentPage} = this.props;
+    const siblings = 1;
+
+    const totalNumbers = siblings + 5;
```
Comment (chilimalone) [inline src/components/Pagination.js:27]: it's not very intuitive what totalnumbers mean

File: .gitignore (hunk 1 above)
Comment (Lord-Bubbles) [reply .gitignore:33]: done

File: src/components/Pagination.js (hunk 3 above)
Comment (Lord-Bubbles) [reply src/components/Pagination.js:27]: added comments explaining this

File: src/components/Pagination.js (hunk 2 above)
Comment (Lord-Bubbles) [reply src/components/Pagination.js:42]: adding comments explaing this

File: src/components/Pagination.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  handleRight = () => {
+    this.props.onPageChange(this.props.currentPage + 1);
+  }
+
+  fetchPageNumbers = () => {
+    const { currentPage, total, pageSize } = this.props;
+    const { siblings } = this.state;
+    const totalPageCount = Math.ceil(total / pageSize);
+
+    // Number of pages to display in pagination bar is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
```
Comment (chilimalone) [inline src/components/Pagination.js:34]: make sure that comments obey 80 character line limit. Break into multiple lines if it doesn't fit

**Review by chilimalone — APPROVED** [review]

File: src/components/Pagination.js (hunk 4 above)
Comment (Lord-Bubbles) [reply src/components/Pagination.js]: done

**Pull request merged**